airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -83,4 +132,347 @@ protected DataType<?> toDialectType(final AirbyteProtocolType airbyteProtocolTyp
 
   protected abstract SQLDialect getDialect();
 
+  /**
+   * @param columns from the schema to be extracted from _airbyte_data column.
+   *                Use the destination specific syntax to extract data
+   * @return a list of jooq fields for the final table insert statement.
+   */
+  protected abstract List<Field<?>> extractRawDataFields(final LinkedHashMap<ColumnId, AirbyteType> columns);
+
+  /**
+   *
+   * @param columns from the schema to be used for type casting errors and construct _airbyte_meta column
+   * @return
+   */
+  protected abstract Field<?> buildAirbyteMetaColumn(final LinkedHashMap<ColumnId, AirbyteType> columns);
+
+  protected abstract Condition cdcDeletedAtNotNullCondition();
+
+  protected abstract Field<Integer> getRowNumber(final List<ColumnId> primaryKey, final Optional<ColumnId> cursorField);
+
+  protected abstract ColumnId getCdcDeletedAtColumn();
```

Comment:
does this work?
```suggestion
  private ColumnId getCdcDeletedAtColumn() {
    return buildColumnId("_ab_cdc_deleted_at");
  }
```

Reply:
I think it should, originally it was an instance variable calling the method which relies on NamingTransformer injected through c'tor and leading to NPE.